### PR TITLE
feat: Allow aborting of requests

### PR DESCRIPTION
**Pull request by jonathandale**

### CODE REVIEW

File: src/re_frame_request/events.cljs [hunk 1]
```diff
@@ -106,32 +127,36 @@
     :keys [name
            on-success
            on-failure
+           on-abort
            on-progress]
     :or {on-success [:http-no-on-success]
          on-failure [:http-no-on-failure]}}]
   (try
-    (let [seq-request-maps (if (sequential? request) request [request])
```
Comment (jonathandale) [inline src/re_frame_request/events.cljs:113]: Not sure how `request` could have been `sequential?` as the destructuring in the function args indicates it's a map, and therefore `name`, `on-success` etc would be missing

File: src/re_frame_request/events.cljs [hunk 2]
```diff
@@ -66,10 +77,20 @@
...
+              (cond-> {:name name
+                       :request-time request-time}
+                (some? request-xhrio)
+                (assoc :abort #(try
+                                 (ajax/abort request-xhrio)
+                                 (when (fn? on-abort)
+                                   (on-abort))
+                                 (catch js/Error e
+                                   (js/console.error (str "Failed to abort request: " e))))))]))
+  ([name request-time]
```
Comment (jonathandale) [inline src/re_frame_request/events.cljs]: Using a multi-arity function for backwards compatibility

File: src/re_frame_request/events.cljs [hunk 3]
```diff
@@ -66,10 +77,20 @@
...
-              :request-time request-time}]))
+  ([{:keys [name request-time request-xhrio on-abort]}]
+   (dispatch [:request/start
+              (cond-> {:name name
+                       :request-time request-time}
+                (some? request-xhrio)
+                (assoc :abort #(try
+                                 (ajax/abort request-xhrio)
+                                 (when (fn? on-abort)
+                                   (on-abort))
```
Comment (jonathandale) [inline src/re_frame_request/events.cljs]: Will dispatch `:on-abort`, if handler in request map

File: src/re_frame_request/events.cljs [hunk 4]
```diff
@@ -57,6 +57,17 @@
...
+(defn wrap-abort!
+  "Wraps the passed in on-abort callback"
+  [on-abort name request-time]
+  (fn []
+    (dispatch [:request/done
+               {:status :aborted
+                :name name
+                :error nil
+                :request-time request-time}])
+    (dispatch (conj on-abort))))
```
Comment (dehli) [inline src/re_frame_request/events.cljs]: Is the `conj` needed?

Comment (jonathandale) [reply src/re_frame_request/events.cljs]: Good catch, no, not needed.

**Review by oconn — APPROVED** [review]
🎉  - Thanks for this work

**Pull request merged**